ocpmodels/datasets/lmdb_dataset.py

```diff
@@ -57,11 +57,21 @@ def __init__(self, config, transform=None):
 
             self._keys, self.envs = [], []
             for db_path in db_paths:
-                self.envs.append(self.connect_db(db_path))
-                length = pickle.loads(
-                    self.envs[-1].begin().get("length".encode("ascii"))
-                )
-                self._keys.append(list(range(length)))
+                cur_env = self.connect_db(db_path)
+                self.envs.append(cur_env)
+
+                # Load and encode all keys in the LMDB
+                cur_keys = [
```

Comment:
Does this increase the time to create the dataset class appreciably for large datasets? I assume not, but it would be good to check, just in case.